### DISCUSSION TITLE
Classes along with style props in every component

### DISCUSSION THREAD

**Discussion by mrhammadasif** [tusen-ai/naive-ui#3926] [Polls]
Almost every component provides a way to provide custom styles (e.g. in n-drawer we have body-style, header-style, etc) but I (and many other developers who use tailwind) really want to provide classes to the elements like (body-class, header-class, etc)

```html
<NDrawer :body-style="{background: 'red'}">
  ...
</NDrawer>
```

when using with TailwindCSS

```html
<NDrawer body-class="bg-red-400 ...">
  ...
</NDrawer>
```

Although, it is not a breaking change, and I am willing to create PR to update but I am looking forward to what you guys think about this.

**Comment by XieZongChen** [comment]
If you start this development, maybe need to consider existing API in some components. And we should discuss a unified naming convention. @07akioni

**Reply by mrhammadasif** [reply]
Yeah, I guess I will be looking for every `x-style` prop and then add a `x-class` prop for each

**Comment by jahnli** [comment]
Now, wherever there is a style, there is a corresponding class
